### PR TITLE
Remove dependency from nose

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by kain88-de** [comment]
Can you remove nose as well from the Travis config. Then we can see if this really is the last thing that uses nose.

**Comment by utkbansal** [comment]
@kain88-de This is not the last thing that uses nose. I'm currently investigating what is still dependent on nose. Will update as I know more.

**Comment by utkbansal** [comment]
@kain88-de https://github.com/MDAnalysis/mdanalysis/pull/1574/files#diff-416947382aa4e650975916732248fef5R519 is failing. I tried using the debugger and the value of `dihedrals[0]` comes out to be `0.0` (float64).

**Comment by kain88-de** [comment]
The diff you are citing is to big.

**Comment by utkbansal** [comment]
@kain88-de `testsuite/MDAnalysisTests/utils/test_distances.py` line 519. Method is `test_dihedrals` in `TestCythonFunctions` class.

**Comment by richardjgowers** [comment]
Not sure why that would have started working, but it looks like the tests pass now?  (Ie it raises the right Error again).  This just needs a rebase now

**Comment by mnmelo** [comment]
This is weird indeed. @nestorwendt's recent PR #1566 went over the behavior of those lines (the one in question seems to be the check whether an undefined dihedral returns a nan).
There should be little numerical instability in this particular check, since it's the case where all four dihedral atoms' positions are [0., 0., 0.].

**Comment by utkbansal** [comment]
@kain88-de @richardjgowers This works on Travis, not on my system. Have tried reinstalling the package and testusite, but the tests still fail.

File: testsuite/MDAnalysisTests/analysis/test_encore.py [hunk 1]
```diff
@@ -364,9 +362,7 @@ def test_ces_error_estimation(self):
         assert_almost_equal(stdev, expected_stdev, decimal=0,
                             err_msg="Unexpected standard daviation  for bootstrapped samples in Clustering Ensemble similarity")
 
-    @dec.skipif(module_not_found('sklearn'),
-                "Test skipped because sklearn is not available.")
-    @dec.slow
+    @pytest.mark.skipif(module_not_found('sklearn'), reason="Test skipped because sklearn is not available.")
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/analysis/test_encore.py]: correct here is [`pytest.importorskip`](https://docs.pytest.org/en/latest/skipping.html#skipping-on-a-missing-import-dependency). Note you need to use it inside of the function.

**Comment by kain88-de** [comment]
@utkbansal I don't know why the test don't work on your local laptop. It's hard to tell without being in front of the computer.

**Comment by utkbansal** [comment]
@kain88-de 

```python
    def chk_same_position(x_id, y_id, hasval='nan'):
        """Handling nan/inf: check that x and y have the nan/inf at the same
            locations."""
        try:
            assert_array_equal(x_id, y_id)
        except AssertionError:
            msg = build_err_msg([x, y],
                                err_msg + '\nx and y %s location mismatch:'
                                % (hasval), verbose=verbose, header=header,
                                names=('x', 'y'), precision=precision)
>           raise AssertionError(msg)
E           AssertionError: 
E           Arrays are not almost equal to 5 decimals
E           Cython dihedrals didn't match numpy calculations
E           x and y nan location mismatch:
E            x: array([ 0.     ,  0.     ,  3.14159,  0.50714])
E            y: array([     nan,      nan,  3.14159, -0.50714], dtype=float32)

```

**Comment by kain88-de** [comment]
Are your cython generated files up do date? We recently changed them so maybe you need to compile the cython modules against the current package version you are testing.

**Comment by utkbansal** [comment]
@kain88-de Can we merge this soon? Would save me from a lot of merge conflicts.

**Pull request merged**